code/modules/ai/ai_behaviors/xeno/puppet.dm

```diff
@@ -181,3 +185,28 @@
 		return
 	if(feed.ai_should_use(target))
 		feed.use_ability(target)
+
+/// rest when puppeter does
+/datum/ai_behavior/puppet/proc/start_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living?.set_resting(TRUE)
+
+/// stop resting when puppeter does, plus unbuckle all mobs so the widow won't get stuck
+/datum/ai_behavior/puppet/proc/stop_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living?.set_resting(FALSE)
+	source?.unbuckle_all_mobs()
```

Comment:
Puppets don't buckle to who they're following, also perhaps make a ai_behavioru subtype that does following, sharing attacks and some actions that could be shared between puppeteer and widow

Reply:
no more unbuckle that was an oversight also I dont know how create a subtype that could be shared between widow and puppeteer. I used the code from widows ai section and her section is coded into the spiderling.dm file .

Reply:
Add a shared type that uses functionality that both could use, and then the widow and puppeteer can have their caste specific code on their own types

Reply:
Im very DM illiterate I actually have no idea how to do that.